project/Build.scala

```diff
@@ -67,9 +68,18 @@ object DottyBuild extends Build {
     // otherwise it just executes the tests directly
     lockPartestFile := {
       val partestLockFile = "." + File.separator + "tests" + File.separator + "partest.lock"
-      partestLock = new RandomAccessFile(partestLockFile, "rw").getChannel.tryLock
+      try {
+        partestLock = new RandomAccessFile(partestLockFile, "rw").getChannel.tryLock
+      } catch {
+        case ex: java.nio.channels.OverlappingFileLockException => // locked already
```

Comment:
This exception is thrown when the lock is held by the same JVM. This could happen if Tests.Cleanup doesn't run for some reason. If the lock is held by another JVM, tryLock returns null. In both cases, we just continue because the file is locked (even though not by us). In general it's a bad idea to have two sbt instances run tests at the same time since we only communicate partest-ness through the file locking, which is a global property, not a per-sbt property. So maybe it would be better to fail if we can't lock the file?

The JUnit test switches between partest file generation and running the tests based on the lock file. So if one sbt instance does `partest` and the other `test`, then the second will still generate partest files (because the file is still locked at the time when the JUnit test runs). So in the second case, the tests are neither run by JUnit nor by partest.